### ISSUE TITLE
possible bug or just confusion about rotation matrix in mask drawing

### ISSUE THREAD

**Issue by mperrin** [spacetelescope/jwst_coronagraph_visibility#18]
Question for you @kjbrooks (and possibly @josePhoenix if you have time to look at this, since you last touched this code.)

I'm looking at this part of the code, which implements the rotation mask for drawing masks on the array: https://github.com/spacetelescope/jwst_coronagraph_visibility/blob/master/jwst_coronagraph_visibility/gui.py#L1243 
and wondering why the rotation matrix there seems wrong to me. It's coded as 
```
cos sin
-sin cos
```
which has the signs of the sins flipped from what I expect. Is there some reason that's correct as-is? 

That code further confuses me because the first line (line 1243, linked above) modifies the x_verts array, and then the _that modified copy_ is what gets used in the second line. So it doesn't apply the rotation matrix correctly. 

For context I encountered this while trying to hack in an implementation of the NIRSpec IFU FOV as a 3rd instrument. In that case with a roughly 45 degree rotation this code is _definitely_ wrong and produces a skew rather than a rotation. I am guessing this is likely a bug in the MIRI code as well, and just has been overlooked until now because cos(4.5 degrees) is nearly 1. What do you think?

**Comment by joseph-long** [comment]
> That code further confuses me because the first line (line 1243, linked above) modifies the x_verts array, and then the that modified copy is what gets used in the second line. So it doesn't apply the rotation matrix correctly.

Yeah, that definitely looks like a bug

**Comment by jordan-stone** [comment]
@mperrin I notice this issue is not closed and the problem of modifying x_verts before defining y_verts using the modified x_verts persists in the code when using MIRI:

https://github.com/spacetelescope/jwst_coronagraph_visibility/blob/bdd0cc19bd86b29b402df524aab634bb7b3d7992/jwst_coronagraph_visibility/gui.py#L1403

https://github.com/spacetelescope/jwst_coronagraph_visibility/blob/bdd0cc19bd86b29b402df524aab634bb7b3d7992/jwst_coronagraph_visibility/gui.py#L1445

Should users be worried?

(Note that sin is an odd function, so swapping the signs of the sins just makes a counterclockwise rotation matrix a clockwise rotation matrix. Which sense of rotation is expected?)

**Comment by jordan-stone** [comment]
Ok, I convinced myself that the bug is real by manually setting y_angle to pi/4 and then looking at the result in the gui. 
https://github.com/spacetelescope/jwst_coronagraph_visibility/blob/bdd0cc19bd86b29b402df524aab634bb7b3d7992/jwst_coronagraph_visibility/gui.py#L1389

This makes the issue dramatic.

**Comment by mperrin** [comment]
Hmm, @jordan-stone thanks for the heads up on this (yikes, a bug report I filed ~6 years ago that is still not fixed?!?). 
But I should note that I have no role in maintaining this package these days.  So let me pass along the FYI instead to @mfixstsci and @aggle

**Comment by aggle** [comment]
Thanks @mperrin for bringing this to our attention

**Comment by aggle** [comment]
@mperrin @jordan-stone @mfixstsci There are two separate issues identified here:

1) the sign convention on the rotation matrix, and 
2) the modification of x_verts before it is used to compute y_verts

2 is a real bug, thanks for submitting the fix @jordan-stone.

I did a sanity check on 1, and is not a bug, as it produces the correct mask representing the orientation of the quadrant boundaries in the MIRI 4QPM masks. 

In the first image, I have attached an on-sky image from the MIRI 1550 coronagraph with the "glowstick" anomaly used to emphasize the orientation of the quadrant boundaries.
<img width="598" alt="image" src="https://github.com/spacetelescope/jwst_coronagraph_visibility/assets/6478178/cea37de1-a9a4-4ce8-96f0-85599dd57a2c">

In the next image, I show the mask orientation of the CVT as it is currently coded:
<img width="479" alt="image" src="https://github.com/spacetelescope/jwst_coronagraph_visibility/assets/6478178/997b2bc0-88a2-4f39-b89b-07915a38e76e">

As you can see, the orientation matches the on-sky data. 

If you reverse the rotation matrix, the mask rotates like so:
<img width="479" alt="image" src="https://github.com/spacetelescope/jwst_coronagraph_visibility/assets/6478178/4705c0e3-414f-46b8-a0dc-ceeba3fd261b">

which is incorrect. 

To summarize, the rotation in the CVT is coded correctly with respect to the definition of the V3IdlYAngle for the aperture in SIAF.
